packages/forms-web-app/src/controllers/final-comment/input-code.js

```diff
@@ -64,15 +73,14 @@ const postInputCode = async (req, res) => {
 	}
 
 	const tokenResult = await isTokenValid(id, token, req.session);
-	// todo: handle too many attempts
 
 	req.session.finalComment.secureCodeEnteredCorrectly = tokenResult.valid;
 
 	if (!req.session.finalComment.secureCodeEnteredCorrectly) {
 		req.session.finalComment.incorrectSecurityCodeAttempts++;
 
 		if (
-			req.session.finalComment.incorrectSecurityCodeAttempts >= finalCommentSecurityCodeMaxAttempts
+			req.session.finalComment.incorrectSecurityCodeAttempts > finalCommentSecurityCodeMaxAttempts
```

Comment:
I've added this logic to the token api to handle attempt counts
tokenResult.tooManyAttempts

Reply:
API returns a 429 and then isTokenValid returns tooManyAttempts: true
The API could potentially do something on the 3rd attempt if wrong.

Reply:
very handy API - have updated the controller.